server/routes/manageConditions/handlers/curfew/curfewRoutes.test.ts

```diff
@@ -0,0 +1,252 @@
+  })
+
+  describe('GET', () => {
+    it('should redirect to additional conditions list page if the additional condition is not found on the licence', async () => {
+      await handler.GET(req, res)
+      expect(res.redirect).toHaveBeenCalledWith('/licence/create/id/1/additional-licence-conditions')
+    })
+
+    it('should redirect to additional conditions list page with fromReviewFlag if the additional condition is not found on the licence', async () => {
+      req.query.fromReview = 'true'
+      await handler.GET(req, res)
+      expect(res.redirect).toHaveBeenCalledWith('/licence/create/id/1/additional-licence-conditions?fromReview=true')
+    })
+
+    it('should format multiple instances of the condition into a single set of form values', async () => {
+      res.locals.licence.additionalLicenceConditions = [curfewInstance1, curfewInstance2] as AdditionalCondition[]
+      const formResponses = {
+        numberOfCurfews: 'Two curfews',
+        curfewStart: SimpleTime.fromString('01:00 am'),
+        curfewEnd: SimpleTime.fromString('02:00 am'),
+        reviewPeriod: 'weekly',
+        numberOfCurfews2: 'Two curfews',
+        curfewStart2: SimpleTime.fromString('04:00 am'),
+        curfewEnd2: SimpleTime.fromString('05:00 am'),
+        reviewPeriod2: 'weekly',
+      }
+
+      await handler.GET(req, res)
+
+      expect(res.render).toHaveBeenCalledWith('pages/manageConditions/curfew/input', {
+        additionalCondition: curfewInstance1,
+        config: conditionConfig,
+        formResponses,
+      })
+    })
+  })
+
+  describe('POST', () => {
+    it('deletes any existing versions of the curfew condtion', async () => {
+      await handler.POST(req, res)
+
+      expect(licenceService.deleteAdditionalConditionsByCode).toHaveBeenCalledWith(
+        CURFEW_CONDITION_CODE,
+        res.locals.licence,
+        res.locals.user
+      )
+    })
+
+    it('adds a number of instances of the curfew condition equal to the user-selection', async () => {
+      jest.spyOn(handler, 'addCurfewCondition')
+      req.body = { numberOfCurfews: 'Three curfews' }
+
+      await handler.POST(req, res)
+
+      expect(handler.addCurfewCondition).toHaveBeenCalledTimes(3)
+    })
+
+    it('should redirect to the callback function', async () => {
+      await handler.POST(req, res)
+      expect(res.redirect).toHaveBeenCalledWith('/licence/create/id/1/additional-licence-conditions/callback')
+    })
+
+    it('should redirect to the callback function with query parameter if fromReview flag is true', async () => {
+      req.query.fromReview = 'true'
+      await handler.POST(req, res)
+      expect(res.redirect).toHaveBeenCalledWith(
+        '/licence/create/id/1/additional-licence-conditions/callback?fromReview=true'
+      )
+    })
+  })
+
+  describe('addCurfewCondition', () => {
+    conditionService.getAdditionalConditionType.mockResolvedValue(LicenceType.AP)
```

Comment:
better to do this within a before each - this is only called once, when the module is imported which means its easily to inadvertently add dependencies in the order in which tests are run.